packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -686,3 +716,35 @@ bool upgradePbxProjWithFlutterAssets(IosProject project, Logger logger) {
   xcodeProjectFile.writeAsStringSync(buffer.toString());
   return true;
 }
+
+void _handleXCResultIssue({required XCResultIssue issue, required Logger logger}) {
+  // Issue summary from xcresult.
+  String issueSummary = '';
+  issueSummary += issue.subType ?? 'Unknown';
+  issueSummary += ' (Xcode): ';
+  issueSummary += issue.message ?? '';
+  issueSummary += issue.location != null? '\n$issue.location' : '';
+
+  switch (issue.type) {
+    case XCResultIssueType.error:
+      logger.printError(issueSummary);
+      break;
+    case XCResultIssueType.warning:
+      logger.printWarning(issueSummary);
+      break;
+  }
+
+  // Add more custom output for flutter users.
+  if (issue.message != null && issue.message!.contains('Provisioning profile')) {
```

Comment:
I think this may also be `No matching provisioning profiles found`, though I didn't remove the profiles on my machine to test. Can you make this case insensitive to make it a little more robust to string changes?
(I lamented to you on chat already that I wish we have error codes to go off of...)